src/main/java/pl/cleankod/exchange/core/usecase/FindAccountAndConvertCurrencyUseCase.java

```diff
@@ -22,22 +26,26 @@ public FindAccountAndConvertCurrencyUseCase(AccountRepository accountRepository,
         this.baseCurrency = baseCurrency;
     }
 
-    public Optional<Account> execute(Account.Id id, Currency targetCurrency) {
+    public Optional<Account> execute(Account.Id id, Currency targetCurrency, String traceId) {
```

Comment:
The whole `core` is tainted with a cross-cutting concern now. I think we should keep the `core` pure from this. Any other ideas how we could have the tracing support for each log entry?